### PR TITLE
Adding resource stabilization time and improved UnexpectedErrorStatus

### PR DESCRIPTION
**Pull request by gabrixdu**
### Adding resource stabilization time and enhancing UnexpectedErrorStatus message

_- Resource stabilization time is a time-delta that resolves the duration
  of a resource to reach a stable state after a stack creation or update._

- This was achieved by timestamping the request-start-time once and
dynamically update the request-in-progress timestamp. The delta will
be resolved every time when a stack operation is invoked.

- Added UnexpectedErrorStatus StackTrace to better identify root causes
  of such errors.

- Adapted unit tests to accept the new changes.

- Added unit test to ensure the stack trace is logged accordingly.

- Added unit test to increase test coverage.

- Testing was done in local account.

### CODE REVIEW

File: aws-rds-cfn-common/src/test/java/software/amazon/rds/common/handler/CommonsTest.java [hunk 1]
```diff
@@ -157,33 +157,27 @@ void test_RootCauseMessage_UnexpectedErrorStatus_Without_text()
...
         final Exception ex = new RuntimeException();
-        final Throwable exRootCause = ExceptionUtils.getRootCause(ex);
+        final String rootCauseExceptionMessage = getRootCauseExceptionMessage(ex);
         final String exceptionClass = ex.getClass().getCanonicalName();
         final ErrorRuleSet ruleSet = ErrorRuleSet.extend(ErrorRuleSet.EMPTY_RULE_SET).build();
+        final StringBuilder test_messageBuilder = new StringBuilder();
         Mockito.doAnswer(invocationOnMock
                 ->{Object arg0 = invocationOnMock.getArgument(0);
-        return null;}).when(requestLogger).log(any(String.class));
+            return null;}).when(requestLogger).log(any(String.class),any(String.class));
```
Comment (dbbh) [inline aws-rds-cfn-common/src/test/java/software/amazon/rds/common/handler/CommonsTest.java:166]: nit: spacing

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java [hunk 2]
```diff
@@ -30,4 +40,24 @@ public boolean isAddTagsComplete() {
...
+    public void timestamp(final String label, final Instant instant) {
+        timestamps.put(label, instant.getEpochSecond());
+    }
+
+    public void timestampOnce(final String label, final Instant instant) {
+        timestamps.computeIfAbsent(label, s -> instant.getEpochSecond());
+    }
+
+    public Instant getTimestamp(final String label) {
+        if (timestamps.containsKey(label) && label != null) {
```
Comment (dbbh) [inline aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java:53]: if (timestamps.containsKey(label) && label != null) {

Comment (dbbh) [reply aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java:53]: Or maybe even `timestamps.contains(label)`

Comment (gabrixdu) [reply aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java:53]: I don't think we need this as there's always labels provided. I'll remove it and check if there's potential challenges that may arise from having that check. I think this

**Review by dbbh — APPROVED** [review]

**Review by moataz-mhmd — APPROVED** [review]

**Pull request merged**